march_gait_selection/src/march_gait_selection/dynamic_gaits/balance_gait.py

```diff
@@ -1,45 +1,40 @@
 from copy import deepcopy
+import os
 import sys
 
 import moveit_commander
 import rospy
 from sensor_msgs.msg import JointState
 from std_msgs.msg import Header
-from visualization_msgs.msg import Marker
 
 from march_shared_classes.gait.joint_trajectory import JointTrajectory
 from march_shared_classes.gait.setpoint import Setpoint
 from march_shared_classes.gait.subgait import Subgait
+from march_shared_resources.srv import CapturePointPose
 
 
 class BalanceGait(object):
     """Base class to create a gait using the moveit motion planning."""
 
+    CAPTURE_POINT_SERVICE_TIMEOUT = 1
+
     def __init__(self, gait_name='gait_balanced_walk', move_groups=None):
         self.gait_name = gait_name
         self.move_group = move_groups
 
         self._default_walk = None
 
         self._end_effectors = {'left_leg': 'foot_left', 'right_leg': 'foot_right'}
-        self._capture_point_pose = {'left_leg': None, 'right_leg': None}
-
-        self._latest_capture_point_msg_time = {'left_leg': None, 'right_leg': None}
-        self._old_capture_point_msg_time = {'left_leg': None, 'right_leg': None}
+        self._capture_point_service = rospy.ServiceProxy('/march/capture_point', CapturePointPose)
 
-        rospy.Subscriber('/march/cp_marker_foot_left', Marker, queue_size=1, callback=self.capture_point_cb,
-                         callback_args='left_leg')
-        rospy.Subscriber('/march/cp_marker_foot_right', Marker, queue_size=1, callback=self.capture_point_cb,
-                         callback_args='right_leg')
+        self._joint_state_target = JointState()
+        self._joint_state_target.header = Header()
+        self._joint_state_target.name = self.move_group['left_leg'].get_active_joints() + self.move_group['right_leg'].\
+            get_active_joints()
```

Comment:
```suggestion
        self._joint_state_target.name = self.move_group['left_leg'].get_active_joints() + 
                                                          self.move_group['right_leg'].get_active_joints()
```
better to split after/before a `+`